frontend/test/pytest/test_preprocess.py

```diff
@@ -582,7 +628,7 @@ def test_controlled_decomposes_to_unitary(self):
         tape = qml.tape.QuantumScript([qml.ctrl(qml.RX(1.23, 0), 1)])
 
         with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
-            (new_tape,), _ = catalyst_decompose(tape, ctx, stopping_condition)
+            (new_tape,), _ = catalyst_decompose(tape, ctx, stopping_condition, capabilities)
 
```

Comment:
Can we add a test similar to this test that checks the behaviour added in this PR, i.e. that whether or not an operator is in the list of ops to decompose to unitary determined whether we decompose it or not?

Reply:
[Done](https://github.com/PennyLaneAI/catalyst/blob/3448527ab3499c984dc782d091ece69978b58789/frontend/test/pytest/test_preprocess.py#L626)